Fix Yocto PROVIDES and remove obsolete recipe.

kacf opened the pull request.

oleorhagen reviewed (approved):
What exactly makes this better? :thinking:

kacf commented:
`PROVIDES += "mender-auth mender-update"` is wrong, since we don't provide multiple recipes for those, only runtime packages. So that one is clear.

The removal of `PROVIDES += "mender"` was intended to get rid of messages of the type "multiple providers are available for...". And at the same time try to minimize the number of `PREFERRED_PROVIDERS` you need to add. However, it's possible it's necessary after all. The build failed and I'm still testing/investigating this.

kacf commented:
Ok, it looks like the latter *is* necessary when building mender-native. We do it in order to include update module generators in our test builds. For users it's probably quite rare, but not impossible, if they need generators in their build. So I'll leave it then.

The downside is that we need even more `PREFERRED_(R)PROVIDER` lines to separate between the two, but oh well.

kacf commented:
@oleorhagen: I've reworked it to include `PROVIDES` again, but I'm using the right `PREFERRED_(R)PROVIDER` defaults to get rid of the warnings. Please have another look!

oleorhagen reviewed (approved):
I honestly thought we wanted the two packages also, so that you can build them independently, but I'm guessing it's not a big deal.

kacf commented:
> I honestly thought we wanted the two packages also, so that you can build them independently, but I'm guessing it's not a big deal.

Careful with terminology: We already have two packages. But I guess you meant two *recipes*?

We still might do that, but not unless we split mender-auth and mender-update more cleanly on the code/repository level. So in other words: I think it would make sense, but not now.

But actually that's exactly why I didn't want to keep any `PROVIDES` because it creates a relationship between `mender` and `mender-client` recipes, which will later be broken if we split them into a pair of `mender-auth`/`mender-update` and `mender-client`. But there seems to be no way to avoid defining `PROVIDES`...

kacf commented on meta-mender-core/classes/mender-setup.bbclass:
```diff
@@ -229,6 +229,10 @@ MENDER_DATA_PART_LABEL ??= "data"
 
 # --------------------------- END OF CONFIGURATION -----------------------------
 
+PREFERRED_PROVIDER_mender-native ??= "mender-native"
```
Oh, the virtual. This got lost during my testing. I recall I had some troubles with it, but let me retry, maybe it was something else.

kacf replied:
Yeah, so the weird thing is: Even though it's documented, using `virtual` doesn't actually work. If I add it the warning comes back:

```
PREFERRED_PROVIDER_virtual/mender-native ??= "mender-native"

...

NOTE: Multiple providers are available for mender-native (mender-native, mender-client-native)
Consider defining a PREFERRED_PROVIDER entry to match mender-native
```
